src/test/java/core/basesyntax/service/process/impl/FruitTransactionServiceImplTest.java

```diff
@@ -0,0 +1,135 @@
+            new FruitTransactionServiceImpl(operationStrategy);
+
+    @BeforeClass
+    public static void beforeClass() {
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.PURCHASE,
+                new PurchaseHandler(storageDao));
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.RETURN,
+                new ReturnHandler(storageDao));
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.BALANCE,
+                new BalanceHandler(storageDao));
+        operationOperationsHandlerMap.put(FruitTransaction.Operation.SUPPLY,
+                new SupplyHandler(storageDao));
+    }
+
+    @After
+    public void tearDown() {
+        Storage.storage.clear();
+    }
+
+    @Test
+    public void processTransactions_balance_ok() {
+        FruitTransaction.Operation balanceOperation = FruitTransaction.Operation.BALANCE;
+        String fruitName = "banana";
+        Integer fruitQuantity = 20;
+        List<FruitTransaction> testData =
+                List.of(new FruitTransaction(balanceOperation, fruitName, fruitQuantity));
+        fruitTransactionService.processTransactions(testData);
+        Integer actualQuantity = Storage.storage.get(fruitName);
+        assertEquals(fruitQuantity, actualQuantity);
+    }
+
+    @Test
+    public void processTransactions_purchase_ok() {
+        FruitTransaction.Operation purchaseOperation = FruitTransaction.Operation.PURCHASE;
+        String fruitName = "banana";
+        Integer startQuantity = 20;
+        Integer purchaseQuantity = 13;
+        Storage.storage.put(fruitName, startQuantity);
+        List<FruitTransaction> testData =
+                List.of(new FruitTransaction(purchaseOperation, fruitName, purchaseQuantity));
+        fruitTransactionService.processTransactions(testData);
+        Integer expectedQuantity = startQuantity - purchaseQuantity;
+        Integer actualQuantity = Storage.storage.get(fruitName);
+        assertEquals(expectedQuantity, actualQuantity);
+    }
+
+    @Test
+    public void processTransactions_purchaseMore_notOk() {
+        FruitTransaction.Operation purchaseOperation = FruitTransaction.Operation.PURCHASE;
+        String fruitName = "banana";
+        Integer startQuantity = 20;
+        Integer purchaseQuantity = 100;
+        Storage.storage.put(fruitName, startQuantity);
+        List<FruitTransaction> testData =
+                List.of(new FruitTransaction(purchaseOperation, fruitName, purchaseQuantity));
+        assertThrows(RuntimeException.class,
+                () -> fruitTransactionService.processTransactions(testData));
+    }
+
+    @Test
+    public void processTransactions_purchaseNonExistent_notOk() {
+        FruitTransaction.Operation purchaseOperation = FruitTransaction.Operation.PURCHASE;
+        String fruitName = "banana";
+        Integer startQuantity = 20;
+        Integer purchaseQuantity = 100;
+        String notValidFruitName = "peach";
```

Comment:
It`s not very good to use this value for notValidFruitName, because in input file (especially in real project) you can get any fruit, in out task "apple" and "banana" was just an example.